interp/constrexpr_ops.ml

```diff
@@ -154,6 +154,7 @@ let rec constr_expr_eq e1 e2 =
     Id.equal id1 id2 && List.equal instance_eq c1 c2
   | CSort(_,s1), CSort(_,s2) ->
     Miscops.glob_sort_eq s1 s2
+  (* Why do we distinguish kinds of casts? *)
   | CCast(_,a1,(CastConv b1|CastVM b1)), CCast(_,a2,(CastConv b2|CastVM b2)) ->
```

Comment:
@maximedenes: A look at the history gives some hints.

The function `constr_expr_eq` is a renaming (9330bf65) of the function `check_same_type` which is used to factorize binders in the `glob_constr` to `constr_expr` translation, as in e.g.:
```coq
Check forall (A : Prop) (p : A) (q : A), p = q.    
(* returns:  forall (A : Prop) (p q : A), p = q *)
```
Commit 38ff8d2b added support for non-factorization of `A : Prop` and `A :>` (which I don't know well).
Commit 5fa47f12 added support for factorization of `A : Prop` and `A <: Prop`.
```coq
Check forall (A : Prop) (p : A : Prop) (q : A <: Prop), p = q.    
(* returns: forall (A : Prop) (p q : A : Prop), A -> B *)
```
After some thinking, my own suggestion would be to backtrack on 5fa47f12 and to consider different ways of casting as non factorizable. So, in practice, the line should rather call `cast_type_eq`. Additionally, this would make the function `constr_expr_eq` closer to the specification that its name suggests. So, two reasons for a change.

Do you share this analysis?